trace_decoder/src/decoding.rs

```diff
@@ -302,11 +295,122 @@ impl ProcessedBlockTrace {
+                    )),
+                    v,
+                )
+            })
+        {
+            slots_nibbles.push(slot);
+
+            // If we are writing a zero, then we actually need to perform a delete.
+            match val == &ZERO_STORAGE_SLOT_VAL_RLPED {
+                false => {
+                    storage_trie.insert(slot, val.clone()).map_err(|err| {
+                        let mut e =
+                            TraceParsingError::new(TraceParsingErrorReason::TrieOpError(err));
+                        e.slot(U512::from_big_endian(slot.bytes_be().as_slice()));
+                        e.slot_value(U512::from_big_endian(val.as_slice()));
+                        e
+                    })?;
+
+                    delta_out
+                        .additional_storage_trie_paths_to_not_hash
+                        .entry(ADDRESS)
+                        .or_default()
+                        .push(slot);
+                }
+                true => {
+                    if let Ok(Some(remaining_slot_key)) =
+                        Self::delete_node_and_report_remaining_key_if_branch_collapsed(
+                            storage_trie,
+                            &slot,
+                        )
+                    {
+                        delta_out
+                            .additional_storage_trie_paths_to_not_hash
+                            .entry(ADDRESS)
+                            .or_default()
+                            .push(remaining_slot_key);
+                    }
+                }
+            };
+        }
+
+        nodes_used.storage_accesses.push((ADDRESS, slots_nibbles));
+
+        let addr_nibbles = Nibbles::from_h256_be(ADDRESS);
+        delta_out
+            .additional_state_trie_paths_to_not_hash
+            .push(addr_nibbles);
+        let addr_bytes = trie_state
+            .state
+            .get(addr_nibbles)
+            .ok_or(TraceParsingError::new(
+                TraceParsingErrorReason::MissingKeysCreatingSubPartialTrie(
+                    addr_nibbles,
+                    TrieType::State,
+                ),
+            ))?;
+        let mut account = account_from_rlped_bytes(addr_bytes)?;
+
+        account.storage_root = storage_trie.hash();
+
+        let updated_account_bytes = rlp::encode(&account);
+        trie_state
+            .state
+            .insert(addr_nibbles, updated_account_bytes.to_vec())
+            .map_err(|err| {
+                let mut e = TraceParsingError::new(TraceParsingErrorReason::TrieOpError(err));
+                e.slot(U512::from_big_endian(addr_nibbles.bytes_be().as_slice()));
+                e
+            })?;
+
+        Ok(())
+    }
+
     fn update_txn_and_receipt_tries(
         trie_state: &mut PartialTrieState,
         meta: &TxnMetaState,
         txn_idx: TxnIdx,
     ) -> TrieOpResult<()> {
+        if meta.txn_bytes.is_none() {
```

Comment:
Minor nit: Wdyt about making `meta.txn_bytes.is_none()` a function called `is_dummy_payload` or similar? This comes up enough in this PR that it might be good for readability (although this is definitely debatable).

Reply:
Yeah sounds good to me!